docs/plans/2026-04-07-provider-improvements.md

```
@@ -0,0 +1,479 @@
+- `Type: "message_start"` with `Model` and `MessageID` fields
+
+#### Tests
+
+- [ ] `TestTurnEvent_InputJsonDelta` — type and text field populated
+- [ ] `TestTurnEvent_MessageStart` — type, model, and message ID populated
+
+---
+
+## Dependency Graph
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)